### ISSUE TITLE
[Feature Request] Alternate CD Samples from Mame2003Plus?

### ISSUE THREAD

**Issue by MavisCelus19201** [finalburnneo/FBNeo#1513]
Libretro's Plus version of MAME 2003 adds in support for alternative CD samples, with games like Final Fight being able to use it's Sega CD OST, with their site listing games like Moonwalker and Out Run among others. Any chance we could see similar expanded sample support in FinalBurn or is that beyond the scope?

**Comment by dinkc64** [comment]
Hi MavisCelus19201,
I think it's a good idea, since we already support this for DonPachi.  Plus I'm working on getting the c64 soundtrack to Ghouls 'n Ghosts hooked up in FBNeo.  Things are going a little slow because of the heatwave lately, though

best regards,
- dink

**Comment by crashGG** [comment]
It would be better if flac or ogg encoding could be added to the CD samples. The uncompressed size of pcm CD samples in mame2003plus is too huge.

**Comment by barbudreadmon** [comment]
Support for ogg/flac samples is unlikely to be added, especially flac which seems to contain GPL code.

**Comment by grant2258** [comment]
No issues with flac plus using plus it has had a full audit from libretro, just some headers and stallman made it clear that simplistic headers doesn't make it a derivative work. So you have no worries that end, if you have any valid concerns with licensing I will  be happy to address them on the 2003-plus repo since I ported it over originally. If you feel there is a violation it should absolutely be addressed.

**Comment by barbudreadmon** [comment]
Ok so what you mean is that none of the GPL parts of libflac are required in the context of decoding ?
Thanks for letting us know.

**Comment by tmaul** [comment]
Libchdr uses this:

https://github.com/rtissera/libchdr/blob/master/include/dr_libs/dr_flac.h

Actually might be a nice way of adding CHD support one day as it only uses zip and pats of 7z and isnt a million extra files to add

**Comment by grant2258** [comment]
> Ok so what you mean is that none of the GPL parts of libflac are required in the context of decoding ?
> Thanks for letting us know.

The only parts that are gpl are some headers not sure what parts your referring too your claiming parts are gpl, the source is in the repo each file has its one licence its not just under gpl. If you do see issues issues it will be removed promptly after investigated if found to be factual.

**Comment by barbudreadmon** [comment]
>your claiming parts are gpl

No, the authors of libflac are the ones claiming some parts of their software is under GPL license : https://xiph.org/flac/license.html

>To be clear this is just regarding plus using it

To be clear, this is the FBNeo repository and i'm not interested in m2k3+'s license. I was not even aware it was using some flac decoding library since @crashGG said it doesn't support flac samples.

I simply asked a question : can you decode flac without using any of the GPL components. Your answer is rather unclear. Well, anyway, i believe we'll simply use the alternative kev mentioned if we ever implement flac support, as this one clearly doesn't include any GPL code.

**Comment by grant2258** [comment]
No sure what you dont understand. Stallman has made it clear simple header files dont make derivative works. Only a few headers are licensed gpl. Its like mame in that way its not all gpl and you can use the original licence in that case when its not gpl. Since the topic mentions plus it did need clarified. Your making no sense here so since its just random comment on your part assuming there is a violation. If you feel its an issue that libretro or myself has missed feel free to open an issue with actual violation and it will be looked into licensing issues and accusation should be deal with and no brushed away.

**Comment by barbudreadmon** [comment]
>I cant help but feel this dev is constantly digging at this core at every opportunity.

You really are a nutcase, i never talked about m2k3+'s license, you are the one talking about it. I don't know if m2k3+ includes GPL code or not, and i don't care.

**Comment by LibretroAdmin** [comment]
See last response here -
https://github.com/libretro/mame2003-plus-libretro/issues/1654

This drama has to stop. Don't fight over it on Github, and not on the subreddit either. If it continues we will have to start moderating. Zero tolerance for pointless drama, we've had all our fill already with that kinda stuff in this scene. We aren't going to tolerate it in our own backyard.

**Comment by grant2258** [comment]
haha class

**Comment by barbudreadmon** [comment]
I removed the pollution, as already stated, whether decoding flac using libflac requires GPL components or not, we aren't likely to use libflac. As for how other projects are handling their license, we never talked about it and it doesn't concern us.

**Comment by dinkc64** [comment]
f* i need to start meditating again

**Comment by mahoneyt944** [comment]
@barbudreadmon @dinkc64  sorry fellas about some miscommunication here, I won't clog up this request past this one post and let you guys get back at it.

It seems since this request referenced support that mame2003-plus has and possible flac support as well ( which grant initially committed in plus) he wanted to be sure that there wasn't an accidental breach of licensing not only for mame2003-plus, but any notion of an implementation here. It doesn't seem there is a violation to worry about in the files used by mame2003-plus as far as I can tell, so you should be able to use it if you'd decide to, however as you pointed out that would not be a concern for your case use if you use an alternative route either. To grants point, if for any reason you would happen to discover any concerns with licensing while implementing something here please tag me so appropriate changes can be made to rectify the situation, but again it doesn't seem like there is any violation which would be concerning to either repository. I hope this clears up some thing, If you have any concerns you wish to discuss you can DM me on discord as well.

Thanks and good luck here 👍

**Comment by dinkc64** [comment]
I'm a nutcase too if that makes anyone feel better.

**Comment by taoenwen** [comment]
We might as well go with this: it's time to end this thread.

**Comment by barbudreadmon** [comment]
Sorry to all nutcases, i shouldn't have compared him to you.

**Comment by dezraj** [comment]
Hello developers! I noticed some games such as SFZ3Mix and DonPachi use alternate sound samples to improve the soundtrack quality in the game. This is also what I wanted for the NeoGeo games which received soundtrack arrangements from SNK. I created a feature request similar to this one, that's why I'm commenting on here. Please take a look:

https://github.com/finalburnneo/FBNeo/issues/1772

Hopefully you'll find time and motivation to implement this request.

Thank you in advance for your time and work.

**Comment by ani917** [comment]
> Libretro's Plus version of MAME 2003 adds in support for alternative CD samples, with games like Final Fight being able to use it's Sega CD OST, with their site listing games like Moonwalker and Out Run among others. Any chance we could see similar expanded sample support in FinalBurn or is that beyond the scope?

@MavisCelus19201 Since these games are hacks and if a similar process is possible, then it is worth trying to contact them
https://www.zeldix.net/
